packages/nostr/src/client/index.ts

```diff
@@ -0,0 +1,297 @@
+import { ProtocolError } from "../error"
+import {
+  EventId,
+  Event,
+  serializeId as serializeEventId,
+  EventKind,
+} from "../event"
+import { PublicKey } from "../keypair"
+import { Conn, IncomingKind, OutgoingKind } from "./conn"
+import * as secp from "@noble/secp256k1"
+import { RawEvent } from "../raw"
+
+/**
+ * A nostr client.
+ */
+export class Nostr {
+  // TODO NIP-44 AUTH, leave this for later
+  /**
+   * Open connections to relays.
+   */
+  readonly #conns: Map<
+    string,
+    {
```

Comment:
Could just make an interface for this to clean it up

Reply:
Hard to pick a name for it. There's already a type called `Conn` and this is just conn with some metadata, but I don't like `ConnMetadata` or `ConnState`. Honestly I find it really neat that TypeScript lets me express this without picking a name for it.